packages/theming/atlas/src/themesource/atlas_core/web/core/widgets/_switch.scss

```diff
@@ -18,6 +18,40 @@ $default-ios-color: rgb(100, 189, 99);
     background-color: lighten($brand-style, 10%);
 }
 
+@mixin style($device, $brand-key, $brand-variable) {
+    .widget-switch-btn-wrapper-#{$brand-key} {
+        .widget-switch {
+            @if ($device == "iOS" or $device == "auto") {
+                &.auto,
+                &.iOS {
+                    .widget-switch-btn-wrapper {
+                        &.checked {
+                            &.widget-switch-btn-wrapper-default {
+                                @include bootstrap-style-ios($brand-variable);
+                            }
+                        }
+                    }
+                }
+            } @else {
+                &.android {
+                    .widget-switch-btn-wrapper {
+                        &.checked {
+                            &.widget-switch-btn-wrapper-default {
+                                @include bootstrap-style-android($brand-variable);
+
+                                .widget-switch-btn {
+                                    background: $brand-variable;
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+
+// maintained for backwards compatibility prior to Switch 2.0.0.
```

Comment:
Does Atlas 3 need to support older widgets?

Reply:
I maintained the emitted selector structure from the previous implementation, in case end-user projects relied on this for any user-defined HTML/widgets.

@cjfhodges what are your thoughts on maintaining these CSS selectors?

Reply:
What does the html structure look like for widget switch? as the class structure seems a bit overkill

Reply:
You see see the HTML structure [here](https://github.com/mendix/widgets-resources/blob/c42159c1e5ce29188d134469098e1e1880478770/packages/pluggableWidgets/switch-web/src/components/Switch.tsx). I think [this](https://github.com/mendix/widgets-resources/pull/559#discussion_r608405251) comment is relevant to this discussion.